src/TParsec/Running.idr

```diff
@@ -37,6 +37,12 @@ MonadRun Maybe where
 MonadRun Identity where
   runMonad (Id a) = pure a
 
+interface Pointed (a : Type) where
+  point : a
+
+Pointed s => MonadRun (StateT s Identity) where
+  runMonad st = pure $ evalState st point
+
```

Comment:
I expect we should be able to define `(MonadRun m, Pointed s) => MonadRun (StateT s m)`